src/cell.rs

```diff
@@ -222,9 +222,9 @@ impl UnitCell {
     /// assert_eq!(cell.angles(), [90.0, 90.0, 90.0]);
     /// assert_eq!(cell.shape(), CellShape::Orthorhombic);
     /// ```
-    pub fn from_matrix(matrix: [[f64; 3]; 3]) -> UnitCell {
+    pub fn from_matrix(mut matrix: [[f64; 3]; 3]) -> UnitCell {
         unsafe {
-            let handle = chfl_cell_from_matrix(matrix.as_ptr());
+            let handle = chfl_cell_from_matrix(matrix.as_mut_ptr());
```

Comment:
Agreed about trying to make the documentation consistent between the different layers!

> Errors in C functions that return a pointer: here one has to take care that one checks for nullptr directly, otherwise this results in a panic, and the error type is unknown.

The error type is unknown, but the error message should always be set. These should only occur if we throw an exception while constructing an object; and these exception have a message.